pkg/aws_config_client/completer.go

```diff
@@ -113,14 +113,99 @@ func (c *completer) SurveyProfile() (*AWSNamedProfile, error) {
 		Name:       profileName,
 		AWSProfile: profile,
 	}
+	consumedProfiles = append(consumedProfiles, namedProfile)
+
+	// delete account from remainingAccounts
+	remainingAccounts[accountIdx] = remainingAccounts[len(remainingAccounts)-1]
+	remainingAccounts = remainingAccounts[:len(remainingAccounts)-1]
+
+	return remainingAccounts, nil
+}
+
+func (c *completer) consumeProfiles(targetRoleName string, accounts []server.AWSAccount) ([]*AWSNamedProfile, []server.AWSAccount) {
+	consumedProfiles := []*AWSNamedProfile{}
+	remainingAccounts := []server.AWSAccount{}
+
+	for _, account := range accounts {
+		profileName := c.calculateDefaultProfileName(account)
+
+		// get the roles associated with this account
+		selectedProfile := server.AWSProfile{}
+		profiles := c.awsConfig.GetProfilesForAccount(account)
+		for _, profile := range profiles {
+			if strings.Contains(profile.RoleARN, targetRoleName) {
+				selectedProfile = profile
+			}
```

Comment:
I feel like we should add the roleName to the profile so you can make direct compares of that instead of relying on the`split` and the `contains` string manipulations. That way we can avoid classes of bugs regarding these string heuristics. 

For example, what if `targetRoleName = foo` but `profile.RoleARN = foobar`? This function would match them even if they are different roles.

Reply:
This is also a case where you'd want to "short circuit" to simplify this logic. 
```
for .... {
    if profile.role != <one we care about> {
        remainingAccounts = append(...)
        continue
    }
    // otherwise do things with this profile
}
```